### PR TITLE
Add aave v2 subgraph support

### PR DESCRIPTION
**Pull request by yabirgb**

### CODE REVIEW

File: docs/changelog.rst [hunk 1]
```diff
@@ -2,11 +2,13 @@
 Changelog
 =========
 
+* :feature:`1857` Premium users will now be able to query Aave V2 events. 
 * :feature:`2722` The sync conflict dialog dates will now be consistent with the user specified date format.
 * :feature:`3114` Users can easily check and manage which addresses are queried for each defi module directly from the respective module page.
 * :feature:`3069` When adding an asset coingecko/cryptocompare identifiers will now be validated and non-existing ones will be rejected.
 * :bug:`2685` Invoking `--version` from the rotki backend binary in Windows should no longer raise a Permission error.
 
+
```
Comment (LefterisJP) [inline docs/changelog.rst:11]: Why the extra space here?

File: rotkehlchen/chain/ethereum/modules/aave/graph.py [hunk 2]
```diff
@@ -581,35 +655,65 @@ def _get_user_data(
...
+                repays += self._parse_repays(user_result['repayHistory'], from_ts, to_ts)
+                liquidation_calls += self._parse_liquidations(
+                    user_result['liquidationCallHistory'],
+                    from_ts,
+                    to_ts,
+                )
+                for key, value in user_result.items():
+                    user_merged_data[key].extend(value)
+
+            if len(query_v2['users']) != 0:
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/modules/aave/graph.py:696]: These 2 ifs are identical. Create a function for it and then use it in both places, don't copy paste the exact same code.

File: rotkehlchen/tests/api/test_aave.py [hunk 3]
```diff
@@ -87,6 +89,26 @@ def test_query_aave_balances(rotkehlchen_api_server, ethereum_accounts):
...
+    """Check querying the aave histoy endpoint works. Uses real data."""
+    rotki = rotkehlchen_api_server.rest_api.rotkehlchen
+    setup = setup_balances(
+        rotki,
+        ethereum_accounts=ethereum_accounts,
+        btc_accounts=None,
+        original_queries=['zerion', 'logs', 'blocknobytime'],
+    )
+
+    _query_simple_aave_history_test_v2(setup, rotkehlchen_api_server, False)
```
Comment (LefterisJP) [inline rotkehlchen/tests/api/test_aave.py:109]:
```suggestion
    )
    _query_simple_aave_history_test_v2(setup, rotkehlchen_api_server, False)
```

unneeded new line

File: rotkehlchen/tests/api/test_aave.py [hunk 4]
```diff
@@ -164,6 +186,48 @@ def _query_simple_aave_history_test(
...
+        if async_query:
+            task_id = assert_ok_async_response(response)
+            # Big timeout since this test can take a long time
+            outcome = wait_for_async_task(server, task_id, timeout=600)
+            assert outcome['message'] == ''
+            result = outcome['result']
+        else:
+            result = assert_proper_response_with_result(response)
+
+    print(result)
```
Comment (LefterisJP) [inline rotkehlchen/tests/api/test_aave.py:210]: remove this
```suggestion
```

**Review by LefterisJP — COMMENTED** [review]
Some first comments. I have not yet tested with my account.

One important note. You should also add something to the DB upgrade. When we are querying aave for an address the used query ranges are saved. So if a user has already queried an address for Aave v1 and upgrades to this code they still wont see aave v2.

You will need to purge all aave history and aave used query ranges in the DB upgradefor things to work correctly.

File: rotkehlchen/chain/ethereum/modules/aave/graph.py [hunk 5]
```diff
@@ -565,6 +639,44 @@ def _process_events(
...
+        withdrawals: List[AaveDepositWithdrawalEvent],
+        borrows: List[AaveBorrowEvent],
+        repays: List[AaveRepayEvent],
+        liquidation_calls: List[AaveLiquidationEvent],
+        user_merged_data: Dict[str, Any],
+        from_ts: Timestamp,
+        to_ts: Timestamp,
+    ) -> None:
+        """
+        Given a query result from the graph (data) this function extracts
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/modules/aave/graph.py:655]: this function extracts ... ?

File: rotkehlchen/chain/ethereum/modules/aave/graph.py [hunk 6]
```diff
@@ -565,6 +639,44 @@ def _process_events(
...
+        """
+        if 'users' in query and len(query['users']) != 0:
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/modules/aave/graph.py:657]: Hmm so if this is `False` nothing happens?

Then please change this function and make it so that if this is `False` the function should exit.

This way the rest of the function will be more readable and less indented.

File: rotkehlchen/chain/ethereum/modules/aave/graph.py [hunk 7]
```diff
@@ -565,6 +639,44 @@ def _process_events(
...
+                    withdrawals=user_result['redeemUnderlyingHistory'],
+                    from_ts=from_ts,
+                    to_ts=to_ts,
+                )
+                borrows += self._parse_borrows(user_result['borrowHistory'], from_ts, to_ts)
+                repays += self._parse_repays(user_result['repayHistory'], from_ts, to_ts)
+                liquidation_calls += self._parse_liquidations(
+                    user_result['liquidationCallHistory'],
+                    from_ts,
+                    to_ts,
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/modules/aave/graph.py:671]: please use keyword arguments

File: rotkehlchen/chain/ethereum/modules/aave/graph.py [hunk 8]
```diff
@@ -581,35 +693,54 @@ def _get_user_data(
...
+            self._process_graph_query_result(
+                query,
+                deposits,
+                withdrawals,
+                borrows,
+                repays,
+                liquidation_calls,
+                user_merged_data,
+                from_ts,
+                to_ts,
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/modules/aave/graph.py:727]: please use keyword args here and below

File: rotkehlchen/db/upgrades/v27_v28.py [hunk 9]
```diff
@@ -18,3 +18,4 @@ def upgrade_v27_to_v28(db: 'DBHandler') -> None:
             'ALTER TABLE yearn_vaults_events ADD COLUMN version INTEGER NOT NULL DEFAULT 1;',
         )
     db.conn.commit()
+    db.delete_aave_data()
```
Comment (LefterisJP) [inline rotkehlchen/db/upgrades/v27_v28.py:21]: Add some mock aave data in the test DB and check they exist and they are deleted after the upgrade

**Review by LefterisJP — COMMENTED** [review]
Some more things needed

File: rotkehlchen/db/upgrades/v27_v28.py (hunk 9 above)
Comment (yabirgb) [reply rotkehlchen/db/upgrades/v27_v28.py:21]: Isn't this testing again that the `delete_aave_data` works?

Comment (LefterisJP) [reply rotkehlchen/db/upgrades/v27_v28.py:21]: no it's testing that you added the aave data in the upgrade. For example ... here you have added the deletion before the commit. You should move it above also.

File: rotkehlchen/chain/ethereum/modules/aave/graph.py [hunk 10]
```diff
@@ -565,6 +639,65 @@ def _process_events(
...
+        except KeyError as e:
+            self.msg_aggregator.add_warning(msg + f' Missing key {str(e)}')
+        try:
+            liquidation_calls += self._parse_liquidations(
+                liquidations=user_result['liquidationCallHistory'],
+                from_ts=from_ts,
+                to_ts=to_ts,
+            )
+        except KeyError as e:
+            self.msg_aggregator.add_warning(msg + f' Missing key {str(e)}')
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/modules/aave/graph.py:697]: Hmm ... why do so many try/catch?

```suggestion
        try:
            deposits += self._parse_deposits(user_result['depositHistory'], from_ts, to_ts)
            withdrawals += self._parse_withdrawals(
                withdrawals=user_result['redeemUnderlyingHistory'],
                from_ts=from_ts,
                to_ts=to_ts,
            )
            borrows += self._parse_borrows(user_result['borrowHistory'], from_ts, to_ts)
            repays += self._parse_repays(user_result['repayHistory'], from_ts, to_ts)
            liquidation_calls += self._parse_liquidations(
                liquidations=user_result['liquidationCallHistory'],
                from_ts=from_ts,
                to_ts=to_ts,
            )
        except KeyError as e:
            self.msg_aggregator.add_warning(msg + f' Missing key {str(e)}')
            return
```

Also .. `return` at the error handling. If a key is missing this is bad, means that the interface has changed so this entry is not to be trusted.

**Review by LefterisJP — COMMENTED** [review]
Small thing needs fixing

File: rotkehlchen/chain/ethereum/modules/aave/graph.py (hunk 10 above)
Comment (yabirgb) [reply rotkehlchen/chain/ethereum/modules/aave/graph.py:697]: I modified it from a simple try/catch to this multiple because if one key is missing but the others are present the job could continue. Now that I see it maybe doesn't make sense. In my head yesterday was a good idea

Comment (LefterisJP) [reply rotkehlchen/chain/ethereum/modules/aave/graph.py:697]: I understand the intention but as you said, definitely does not make sense. Please change it.

File: rotkehlchen/chain/ethereum/modules/aave/graph.py [hunk 11]
```diff
@@ -565,6 +639,53 @@ def _process_events(
...
+        - borrows
+        - repays
+        - liquidation_calls
+        and extends the corresponding arguments with the obtained information.
+        """
+        if 'users' not in query or len(query['users']) == 0:
+            # If there is no information on the query finish the execution
+            log.debug(f'Aave subgraph query has no information for user. {str(query)}')
+            return
+        user_result = query['users'][0]
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/modules/aave/graph.py:667]: nitpick: I say all the time you are putting a lot of unnecessary whitespace. But here, you can be. It separates a section. Makes the return a lot more visible.
```suggestion
            return
        user_result = query['users'][0]
```

File: rotkehlchen/chain/ethereum/modules/aave/graph.py [hunk 12]
```diff
@@ -565,6 +639,53 @@ def _process_events(
...
+            repays += self._parse_repays(user_result['repayHistory'], from_ts, to_ts)
+            liquidation_calls += self._parse_liquidations(
+                liquidations=user_result['liquidationCallHistory'],
+                from_ts=from_ts,
+                to_ts=to_ts,
+            )
+        except KeyError as e:
+            self.msg_aggregator.add_warning(msg + f' Missing key {str(e)}')
+            return
+        for key, value in user_result.items():
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/modules/aave/graph.py:686]: nitpick: same here

```suggestion
            self.msg_aggregator.add_warning(msg + f' Missing key {str(e)}')
            return

        for key, value in user_result.items():
```

**Review by LefterisJP — APPROVED** [review]
LGTM. From the e2e tests that failed it seems like a hiccup in adding the monero manual balance. If it happens again we should check it out.
